coordinates/source/dhtslib/package.d

```diff
@@ -342,75 +357,60 @@ struct Interval(CoordSystem cs)
         }
     }
 
-    auto getMinStart(Interval!cs other)
+    /// get minimum start value between this interval
+    /// and other interval
+    auto getMinStart(Interval!cs other) @nogc
     {
         return this.start < other.start ? this.start : other.start;
     }
 
-    auto getMaxStart(Interval!cs other)
+    /// get maximum start value between this interval
+    /// and other interval
+    auto getMaxStart(Interval!cs other) @nogc
     {
         return this.start > other.start ? this.start : other.start;
     }
 
-    auto getMinEnd(Interval!cs other)
+    /// get minimum end value between this interval
+    /// and other interval
+    auto getMinEnd(Interval!cs other) @nogc
     {
         return this.end < other.end ? this.end : other.end;
     }
     
-    auto getMaxEnd(Interval!cs other)
+    /// get maximum end value between this interval
+    /// and other interval
+    auto getMaxEnd(Interval!cs other) @nogc
     {
         return this.end > other.end ? this.end : other.end;
     }
 
     /// set operators for intersect, union, and difference
-    auto opBinary(string op)(Interval!cs other)
+    auto opBinary(string op)(Interval!cs other) @nogc
+    if(op == "|" || op == "&")
     {
         static if(op == "|") return unionImpl(other);
         else static if(op == "&") return intersectImpl(other);
         else static assert(0,"Operator "~op~" not implemented");
     }
 
+    /// opbinary + or - to use offset
+    auto opBinary(string op, T)(T off) @nogc
+    if ((op == "+" || op == "-") && isIntegral!T)
+    {
+        static if(op == "+")
+            return this.offset(off);
+        else static if(op == "-")
+            return this.offset(-off);
+        else static assert(0,"Operator "~op~" not implemented");
+    }
+
     /// Get string representation for printing
     string toString() const{
         return "[" ~ CoordSystemLabels[cs] ~ "] " ~ this.start.pos.to!string ~ "-" ~ this.end.pos.to!string;
```

Comment:
It's not clear to me that enum indexing like this works, is this left over from when you had teh static array?

In any case, what about
```
cs.stringof.split('.')[$ - 1] or without use of std.array.split: cs.stringof[CoordSystem.stringof.length + 1 .. $].
```

Reply:
Funny enough I am not sure how it works either but it does. Our unit tests do write out a line that says `[obc]100-100` using the `__traits` implementation. 

As for the other part, that was his suggestion for another way to do it.